File: database/convert_json_to_sql.py
import json

json_file = 'c:/laragon/www/crm360/documentos/listado_empresas_filtrado.json'
sql_file = 'c:/laragon/www/crm360/database/import_companies.sql'

with open(json_file, 'r', encoding='utf-8') as f:
    data = json.load(f)

with open(sql_file, 'w', encoding='utf-8') as out:
    out.write('USE crm360;\n\n')
    for row in data:
        ruc = str(row.get('ruc', '')).replace("'", "''")
        business = str(row.get('business_name', '')).replace("'", "''")
        trade = str(row.get('trade_name', '')).replace("'", "''")
        address = str(row.get('parent_address', '')).replace("'", "''")
        mobile = str(row.get('mobile', '')).replace("'", "''")
        category = str(row.get('category_company', '')).replace("'", "''")
        date = str(row.get('date_creation', ''))
        
        sql = f"INSERT IGNORE INTO `companies` (`ruc`, `business_name`, `trade_name`, `address`, `mobile`, `category`, `creation_date`) VALUES ('{ruc}', '{business}', '{trade}', '{address}', '{mobile}', '{category}', '{date}');\n"
        out.write(sql)

print(f"Successfully generated {sql_file}")

File: database/convert_json_to_sql.php
<?php

$json_file = 'c:/laragon/www/crm360/documentos/listado_empresas_filtrado.json';
$sql_file = 'c:/laragon/www/crm360/database/import_companies.sql';

$json_data = file_get_contents($json_file);
$data = json_decode($json_data, true);

if ($data === null) {
    die("Error decoding JSON\n");
}

$sql_content = "USE crm360;\n\n";

foreach ($data as $row) {
    $ruc = str_replace("'", "''", $row['ruc'] ?? '');
    $business = str_replace("'", "''", $row['business_name'] ?? '');
    $trade = str_replace("'", "''", $row['trade_name'] ?? '');
    $address = str_replace("'", "''", $row['parent_address'] ?? '');
    $mobile = str_replace("'", "''", $row['mobile'] ?? '');
    $category = str_replace("'", "''", $row['category_company'] ?? '');
    $date = $row['date_creation'] ?? '';

    $sql = "INSERT IGNORE INTO `companies` (`ruc`, `business_name`, `trade_name`, `address`, `mobile`, `category`, `creation_date`) VALUES ('$ruc', '$business', '$trade', '$address', '$mobile', '$category', '$date');\n";
    $sql_content .= $sql;
}

file_put_contents($sql_file, $sql_content);
echo "Successfully generated $sql_file\n";
